v2/kafka-to-gcs/src/main/java/com/google/cloud/teleport/v2/templates/KafkaToGCS.java

```diff
@@ -0,0 +1,264 @@
+ *     }
+ *  }
+ *
+ * # Execute template:
+ * API_ROOT_URL="https://dataflow.googleapis.com"
+ * TEMPLATES_LAUNCH_API="${API_ROOT_URL}/v1b3/projects/${PROJECT}/templates:launch"
+ * JOB_NAME="kafka-to-gcs-`date +%Y%m%d-%H%M%S-%N`"
+ *
+ * time curl -X POST -H "Content-Type: application/json"     \
+ *     -H "Authorization: Bearer $(gcloud auth print-access-token)" \
+ *     "${TEMPLATES_LAUNCH_API}"`
+ *     `"?validateOnly=false"`
+ *     `"&dynamicTemplate.gcsPath=${BUCKET_NAME}/path/to/image-spec"`
+ *     `"&dynamicTemplate.stagingLocation=${BUCKET_NAME}/staging" \
+ *     -d '
+ *      {
+ *       "jobName":"'$JOB_NAME'",
+ *       "parameters": {
+ *           "bootstrapServers":"broker_1:9092,broker_2:9092",
+ *           "inputTopics":"topic1,topic2",
+ *           "outputDirectory":"'$BUCKET_NAME/path/to/output-location'",
+ *           "outputFileFormat":"text",
+ *           "outputFilenamePrefix":"output",
+ *           "windowDuration":"5m",
+ *           "numShards":"5"
+ *        }
+ *       }
+ *      '
+ * </pre>
+ */
+public class KafkaToGCS implements WriteToGCS {
+
+  /* Logger for class. */
+  private static final Logger LOG = LoggerFactory.getLogger(KafkaToGCS.class);
+
+  /**
+   * The {@link KafkaToGCSOptions} class provides the custom execution options passed by the
+   * executor at the command-line.
+   */
+  public interface KafkaToGCSOptions extends PipelineOptions {
+    @Description("Kafka Bootstrap Servers. ")
+    @Required
+    String getBootstrapServers();
+
+    void setBootstrapServers(String bootstrapServers);
+
+    @Description("Kafka topic to read the input from. ")
+    @Required
+    String getInputTopics();
+
+    void setInputTopics(String inputTopics);
+
+    @Description("The directory to output files to. Must end with a slash. ")
+    @Required
+    String getOutputDirectory();
+
+    void setOutputDirectory(String outputDirectory);
+
+    @Description(
+        "The format of the file to write to. Text file is the Default File Format. "
+            + "Allowed formats are: "
+            + "TEXT, AVRO, PARQUET")
+    @Default.String("text")
+    String getOutputFileFormat();
+
+    void setOutputFileFormat(String outputFileFormat);
+
+    @Description(
+        "The filename prefix of the files to write to. Default file prefix is set to \"output\". ")
+    @Default.String("output")
+    String getOutputFilenamePrefix();
+
+    void setOutputFilenamePrefix(String outputFilenamePrefix);
+
+    @Description(
+        "The window duration in which data will be written. Defaults to 5m. "
+            + "Allowed formats are: "
+            + "Ns (for seconds, example: 5s), "
+            + "Nm (for minutes, example: 12m), "
+            + "Nh (for hours, example: 2h).")
+    @Default.String("5m")
+    String getWindowDuration();
+
+    void setWindowDuration(String windowDuration);
+
+    @Description(
+        "The maximum number of output shards produced when writing. Default set to 0 shards. ")

```

Comment:
Default set to 0 shards sounds confusing. We should say, default is runner defined.